### ISSUE TITLE
[Machine Translation] Most recent download / PR creation workflow failed for MT content

### ISSUE THREAD

**Issue by nr-kkenney** [newrelic/docs-website#5485]
## Summary
Most recent run of the job which PRs machine translated content failed, which can be seen [here](https://github.com/newrelic/docs-website/runs/4701678554?check_suite_focus=true). 

Error:
```none
Error: Invalid or unsupported zip format. No END header found
    at readMainHeader (/home/runner/work/docs-website/docs-website/node_modules/adm-zip/zipFile.js:107:10)
    at new module.exports (/home/runner/work/docs-website/docs-website/node_modules/adm-zip/zipFile.js:23:3)
    at new module.exports (/home/runner/work/docs-website/docs-website/node_modules/adm-zip/adm-zip.js:25:10)
    at fetchAndDeserialize (/home/runner/work/docs-website/docs-website/scripts/actions/fetch-and-deserialize.js:79:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 0)
    at async main (/home/runner/work/docs-website/docs-website/scripts/actions/check-job-progress.js:139:5)
Error: Process completed with exit code 1.
```

## Acceptance Criteria
- [ ] Fix issue so job completes

**Comment by nr-kkenney** [comment]
EOD progress update:

I merged a PR in to fix a couple other awkward things I noticed when attempting to fix the heart of this ticket. For this ticket specifically, Clinton and I now have a pretty good guess as to why this is failing and some next steps.

The request URI we are using to download our files is roughly ~177000 characters long, which is probably not cool for using with fetch -- @roadlittledawn did a quick search and saw a 80kb limit for URI length + request headers -- and why we aren't downloading anything successfully. Additionally, its likely that a number of the files even when downloaded are going to run into deserialization errors -- we saw this for at least 1 file, so that is the next thing to tackle.

But for tomorrow:
* split downloading files into multiple requests rather than all at once
* when deserializing files, don't fail one fail all. do as many as we can, and write out the errors.

**Comment by nr-kkenney** [comment]
checklist of work for sanity:

- [ ] 0a. batch processing of file downloads.

- [ ] 0b. individual processing of file serialization (rather than as batches).

- [ ] 1a. For each file, download, deserialize, set DB record to complete if succesful. If error is encountered (only in deserialization?), set record to errored.

For each file, return its status -- errored or completed, etc. This allows us to aggregate the data.

- [ ] 1b. If an errored record exists, say for japanese hello_world.txt, dont allow for submitting further translations until we have addressed the error.

- [ ] 2a. If a job has no failed files, set its state to COMPLETE. If it has errors, set its state to COMPLETE_WITH_ERRORS indicating processing is complete, and that the errors will be handled adhoc and addressed in a subsequent new job.

- [ ] 2b. Don't delete errored jobs, but delete translations_jobs records for errored jobs.

- [ ] 3. If there is at least one failure, fail the workflow (we still want a PR).

**Comment by nr-kkenney** [comment]
addressed all but one of the PR comments. 

[this comment](https://github.com/newrelic/docs-website/pull/5568/files#r782611254) in particular, calls out a specific important edge that previously, was actually handled w/o trouble, but with the changes we've introduced, now isn't handled 🙈 

the tl;dr of the edge case - after we submit say 10 files for translation, julien may decide to cancel / remove a file from being translated via the vendor UI. im pretty sure this has happened before, so the likelihood of this occurring is reasonable in my mind.

previously, because of the all or nothing nature of the workflow and the way we structured updating the state of jobs & translations, this edge case was irrelevant / handled.

however, since we are now explicitly counting and comparing records in our DB to files correctly deserialized from the vendor, we need to account for this possibility since this could result in endlessly running jobs -- if we dont download a translation (because we stopped translating it) it wont complete.

so how do we get around this?

the simplest thing that comes to mind is to look for cancelled files as one of the earlier steps in `check-job-progress.js`, and delete its translation record.

this also pushes me further in the direction of another improvement i wanted to make process wise, which is to query the DB for translations in_progress, rather than querying the vendor. But that can be saved for another time.

Hopefully this provides enough context if someone picks this branch up.
